include/envoy/event/scaled_timer_minimum.h

```diff
@@ -0,0 +1,56 @@
+#pragma once
+
+#include "absl/types/variant.h"
+
+namespace Envoy {
+namespace Event {
+
+/**
+ * Describes a minimum timer value that is equal to a scale factor applied to the maximum.
+ */
+struct ScaledMinimum {
+  explicit ScaledMinimum(double scale_factor) : scale_factor_(scale_factor) {}
+  const double scale_factor_;
+};
+
+/**
+ * Describes a minimum timer value that is an absolute duration.
+ */
+struct AbsoluteMinimum {
```

Comment:
no unit test for these?

Reply:
What's to unit test? There are tests to check that ScaledTimerMinimum does the right thing since that's where all the logic is. Also #13800 adds some equality checks.

Reply:
Oh I missed that this is not new code, but you are just moving it, so there are existing tests. Sorry about that.